### PR TITLE
♻️  `webserver`: fixes mypy issues in `redis` & `products_middlewares` plugins 

### PR DESCRIPTION
**Pull request by matusdrobuliak66**
## What do these changes do?
- fixes mypy issues in `webserver`'s ``redis`` & ``products_middlewares`` plugin

## Related issue/s

- related to #4071

### CODE REVIEW

File: services/web/server/src/simcore_service_webserver/redis.py [hunk 1]
```diff
@@ -64,8 +63,8 @@ async def _create_client(address: str) -> aioredis.Redis:
                     f"{client=}",
                     json.dumps(attempt.retry_state.retry_object.statistics),
                 )
-            assert client  # nosec
-            return client
+        assert client  # nosec
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/redis.py:66]: actually you should indent them inside the ``with attempt:`` , right?

**Review by pcrespov — APPROVED** [review]
thx!

**Review by sanderegg — APPROVED** [review]

File: services/web/server/src/simcore_service_webserver/products_middlewares.py [hunk 2]
```diff
@@ -1,24 +1,25 @@
...
 
 
-def discover_product_by_hostname(request: web.Request) -> Optional[str]:
-    for product in request.app[APP_PRODUCTS_KEY].values():
+def discover_product_by_hostname(request: web.Request) -> str | None:
+    products: dict[str, Product] = request.app[APP_PRODUCTS_KEY]
+    for _, product in products.items():
         if product.host_regex.search(request.host):
-            return product.name
+            return f"{product.name}"
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/products_middlewares.py:17]: If `product.name` is already a string, this does the job. Not sure what is better though.
```suggestion
            product_name: str = product.name
            return product_name
```

File: services/web/server/src/simcore_service_webserver/redis.py [hunk 3]
```diff
@@ -64,8 +63,8 @@ async def _create_client(address: str) -> aioredis.Redis:
...
+        return client
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/redis.py:67]: This looks wrong, but this code will get removed by https://github.com/ITISFoundation/osparc-simcore/pull/4055

**Review by GitHK — APPROVED** [review]

File: services/web/server/src/simcore_service_webserver/redis.py (hunk 3 above)
Comment (matusdrobuliak66) [reply services/web/server/src/simcore_service_webserver/redis.py]: yes already pushed corrected version :+1:

File: services/web/server/src/simcore_service_webserver/products_middlewares.py (hunk 2 above)
Comment (matusdrobuliak66) [reply services/web/server/src/simcore_service_webserver/products_middlewares.py]: thanks! modified it and is working

File: services/web/server/src/simcore_service_webserver/redis.py (hunk 1 above)
Comment (matusdrobuliak66) [reply services/web/server/src/simcore_service_webserver/redis.py]: yes correct thanks! modified it and added raise exception so mypy doesnt complain about not returning statemant.

**Pull request merged**